src/components/ChatUserList.tsx

```diff
@@ -0,0 +1,34 @@
+import getAllUsers from "@/data-access/users/get-all-users";
+import React, { useEffect, useState } from "react";
+import ChatUser from "./chatuser";
+import { BookUser } from "@/types/user";
+
+const ChatUserList = () => {
+  const [users, setUsers] = useState<BookUser[]>([]);
+
+  useEffect(() => {
```

Comment:
does it work if instead of the useEffect you just have
`const users = await getAllUsers()`

Reply:
you can't use use effect with client components